Reapply "[jit] Implement ScriptProfile to collect instruction profiles."

zhxchen17 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #57398 [jit] Add Python API for ScriptProfile
* **#58783 Reapply "[jit] Implement ScriptProfile to collect instruction profiles."**

This reverts commit fc804b5def5e7d7ecad24c4d1ca4ac575e588ae8.

Differential Revision: [D28617037](https://our.internmc.facebook.com/intern/diff/D28617037)

zhxchen17 committed:
Reapply "[jit] Implement ScriptProfile to collect instruction profiles."

This reverts commit fc804b5def5e7d7ecad24c4d1ca4ac575e588ae8.

[ghstack-poisoned]

zhxchen17 commented:
@zhxchen17 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28617037).

gmagogsfm commented on torch/csrc/jit/runtime/script_profile.cpp:
```diff
@@ -0,0 +1,177 @@
+  return nullptr;
+}
+
+const auto torchBindInitializer = initBindings();
```
I think this is fine.

zhxchen17 committed:
Update on "Reapply "[jit] Implement ScriptProfile to collect instruction profiles.""


This reverts commit fc804b5def5e7d7ecad24c4d1ca4ac575e588ae8.

Differential Revision: [D28617037](https://our.internmc.facebook.com/intern/diff/D28617037)

[ghstack-poisoned]

zhxchen17 commented:
@zhxchen17 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28617037).
